### ISSUE TITLE
Specify configuration options via factory method

### ISSUE THREAD

**Issue by Splines** [eslint-stylistic/eslint-stylistic#254]
### Clear and concise description of the problem

The [`quotes`](https://eslint.style/rules/default/quotes#options) rule states it has two options: a string option and an object option. From the documentation, I don't understand how to specify that I want to use double quotes _and_ have the `avoidEscape` set to true via the [`ConfigurationFactory`](https://eslint.style/guide/config-presets#configuration-factory).

For example,

```js
const customizedStylistic = stylistic.configs.customize({
  "quotes": ["double", { avoidEscape: true }],
});
```

fails with "Value ... should be a string". Note that I have this in my `module.exports`:
```js
rules: {
    ...customizedStylistic.rules,
  }
```

### Suggested solution

What works for me is to directly specify the rule in `module.exports`:
```js
rules: {
    ...customizedStylistic.rules,
    "@stylistic/quotes": ["error", "double", { avoidEscape: true }],
  },
```

Is there a way to have the `avoidEscape` directly in the configuration factory?

_As this workaround gets the job done, this is not at all an urgent issue._

### Validations

- [X] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### Contributes

- [ ] If this feature request is accepted, I am willing to submit a PR to fix this issue
